test/lib/fs.readdir.spec.js

```diff
@@ -81,6 +82,66 @@ describe('fs.readdir(path, callback)', function() {
       }
     );
   });
+
+  inVersion('>=10.10').it('should support "withFileTypes" option', function(
+    done
+  ) {
+    fs.readdir(
+      path.join('nested', 'sub', 'dir'),
+      {withFileTypes: true},
+      function(err, items) {
+        assert.isNull(err);
+        assert.isArray(items);
+        assert.deepEqual(items, [
+          {name: 'empty'},
+          {name: 'one.txt'},
+          {name: 'two.txt'}
+        ]);
+        assert.ok(items[0].isDirectory());
+        assert.ok(items[1].isFile());
+        assert.ok(items[2].isFile());
+        done();
+      }
+    );
+  });
+
+  withPromise.it('should support "withFileTypes" option', function(done) {
+    fs.promises
+      .readdir(path.join('nested', 'sub', 'dir'), {withFileTypes: true})
+      .then(function(items) {
+        assert.isArray(items);
+        assert.deepEqual(items, [
+          {name: 'empty'},
+          {name: 'one.txt'},
+          {name: 'two.txt'}
+        ]);
+        assert.ok(items[0].isDirectory());
+        assert.ok(items[1].isFile());
+        assert.ok(items[2].isFile());
+        done();
+      }, done);
+  });
+
+  inVersion('>=10.10').it(
+    'should support "withFileTypes" option with "encoding" option',
+    function(done) {
+      fs.readdir(
+        path.join('nested', 'sub', 'dir'),
+        {withFileTypes: true, encoding: 'buffer'},
+        function(err, items) {
+          assert.isNull(err);
+          assert.isArray(items);
+          items.forEach(function(item) {
+            assert.equal(Buffer.isBuffer(item.name), true);
+          });
```

Comment:
Please test all the 3 buffer content.
Please add a duplicated withPromise test too.

Reply:
Thanks for the code review 🎉 
Tests added.